layer.go

```diff
@@ -146,5 +147,17 @@ func (l *Layers) Layer(name string) (Layer, error) {
 		return Layer{}, fmt.Errorf("unable to decode layer metadata %s\n%w", f, err)
 	}
 
+	if !layer.Build && !layer.Cache && !layer.Launch {
+		// if all three are false, that could mean we have a API <= 0.5 TOML file
+		// try parsing the <= 0.5 API format where these were top level attributes
+		buf := internal.LayerAPI5{}
+		if _, err := toml.DecodeFile(f, &buf); err != nil && !os.IsNotExist(err) {
+			return Layer{}, fmt.Errorf("unable to decode layer metadata %s\n%w", f, err)
```

Comment:
Given that we are doing a best effort decode here, should we be propogating this error upwards or should we just ignore it?

Reply:
When I was playing around with this, it seems like the 0.6 API layer metadata struct can be used with parsing either the layer 0.5 or 0.6 format TOML. It's doesn't read the correct values, but it's structurally fine and the TOML library won't throw an error.

What's broken is that if the API 0.6 struct is used to parse the layer 0.5 TOML (or vise versa), you'll miss the build, launch, cache properties because they are in a different place. It doesn't cause an error, I think because we're not using strict parsing, but those fields will all end up being the default value, hence my use of build=false, launch=false, cache=false as the trigger to try parsing using the 0.5 API struct.

I think we should propagate up errors for both parsing attempts because, as far as I can tell, a property formatted layer TOML file shouldn't cause either parsing attempt to fail and return an error. If there was an error, I think it would put libcnb into a state where we shouldn't continue. If we just logged the error and continued on, it could mean that build=false, launch=false, cache=false is set set for a layer where the metadata is supposed to indicate something else.

Anyway, that's my thought process. I'm open to suggestions if you think there's a situation where we could benefit from logging the error and continuing processing.

Reply:
That makes sense - the only thing that slightly bugs me is that fact that the layer flags are always set to false starting API 0.6 per https://buildpacks.io/docs/reference/spec/migration/buildpack-api-0.5-0.6/#opt-in-layer-caching-and-launchtoml-format-changes and this was done to avoid issues where leftover layers were being persisted even after they were removed from the buildpack code. In the spirit of this RFC does it make sense to avoid doing this decode entirely or should we still attempt to do it as it was part of the v0.5 api. The other thing to note is that the v0.6+ case will always hit this condition and attempt to decode it using the old api.

I guess one alternative (which may not be worth the effort) is to make `Layers`, BP API aware when we construct it in `BuildContext` and have it decode it using this internal struct only when the API version is 0.5. LMK what you think.

Reply:
>In the spirit of this RFC does it make sense to avoid doing this decode entirely or should we still attempt to do it as it was part of the v0.5 API.

I don't think this PR changes the behavior that we currently have with the 0.6 API support. If you have a 0.6 API buildpack and it sets launch=false, build=false, cache=false on a layer, that'll still be what get's picked. The only difference is that there will be a second attempt to reach the 0.5 API layer. In that case, if this is legitimately a 0.6 API layer with launch=false, build=false, cache=false will come back false because the API 0.6 layer TOML file won't have the properties in the right location to be parsed into the 0.5 API struct. We do assign them back to the struct, but they'll end up all being false so in the end launch=false, build=false and cache=false.

I put this together mostly as a way to get things working for now. Long term, if we are going to support multiple API layers, what we need is an easier way to access the API version, that way it can decode specifically for the API being used and not have to guess.

My preference would be to handle that as part of #48 and go with this for now. Would that be OK?

Reply:
SGTM!